### PR TITLE
Release notes for 8.12 + update to Reporting Problems page

### PR DESCRIPTION
**Pull request by lkasso**
### Details
This is a Documentation change ONLY.

### Context
- Update release notes for Gradle 8.12
- https://github.com/gradle/gradle/issues/31657
- https://github.com/gradle/gradle/issues/31664

### CODE REVIEW

File: platforms/documentation/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -218,6 +219,6 @@ We love getting contributions from the Gradle community. For information on cont
 ## Reporting problems
 
 If you find a problem with this release, please file a bug on [GitHub Issues](https://github.com/gradle/gradle/issues) adhering to our issue guidelines.
-If you're not sure you're encountering a bug, please use the [forum](https://discuss.gradle.org/c/help-discuss).
+If you're not sure if you're encountering a bug, please use the [forum](https://discuss.gradle.org/c/help-discuss).
```
Comment (ov7a) [inline platforms/documentation/docs/src/docs/release/notes.md:222]: 💅 probably should be fixed in the template as well

File: platforms/documentation/docs/src/docs/userguide/authoring-builds/plugins/reporting_problems.adoc [hunk 2]
```diff
@@ -95,3 +95,26 @@ The progress listener callback should then check if the operation result is of t
 
 In addition, there's a more convenient way to access the failure details.
 If clients configure the project connection with `LongRunningOperation.withFailureDetails()`, the Tooling API implicitly subscribes to the `ROOT` operation type and provides failure details via the `GradleConnectionException.getFailures()` method.
+
+[[sec:generated_html_report]]
+=== Generated HTML report
```
Comment (reinsch82) [inline platforms/documentation/docs/src/docs/userguide/authoring-builds/plugins/reporting_problems.adoc:100]: @donat WDYT?

File: platforms/documentation/docs/src/docs/release/notes.md (hunk 1 above)
Comment (lkasso) [reply platforms/documentation/docs/src/docs/release/notes.md:222]: Done.

File: platforms/documentation/docs/src/docs/userguide/authoring-builds/plugins/reporting_problems.adoc (hunk 2 above)
Comment (donat) [reply platforms/documentation/docs/src/docs/userguide/authoring-builds/plugins/reporting_problems.adoc:100]: I'm happy with the content, but the screenshot is not right. It is only a demonstration for the problem summarisation.

You can probably find a better example amongst the Gradle CI builds.

**Pull request merged**